lib/index.ts

```diff
@@ -140,8 +192,68 @@ async function buildSubTreeRecursive(
     }
     // dependency was not found on a current path, remove last key (move closer to the root) and try again
     // visitedDepPaths can be passed by a reference, because traversing up doesn't update it
-    return buildSubTreeRecursive(depName, lockfilePath.slice(0, -1), lockFile, depPath);
+    return buildSubTreeRecursiveFromPackageLock(depName, lockfilePath.slice(0, -1), lockFile, depPath);
+  }
+}
+
+async function buildSubTreeRecursiveFromYarnLock(
+  searchedDep: Dep, lockFile: YarnLock, depPath: string[] ): Promise<PkgTree> {
+  const depSubTree: PkgTree = {
+    depType: searchedDep.dev ? DepType.dev : DepType.prod,
+    dependencies: {},
+    name: searchedDep.name,
+    version: undefined,
+  };
+
+  const depKey = `${searchedDep.name}@${searchedDep.version}`;
+
+  const dep = _.get(lockFile.object, depKey);
+
+  if (!dep) {
+    throw new Error(`Dependency ${depKey} was not found in yarn.lock.
+      Your package.json and yarn.lock are probably out of sync.
+      Please run npm install and try to parse the log again.`);
```

Comment:
This sentence doesn't quite make sense to me: `Please run npm install and try to parse the log again.`

Reply:
typo 😇